streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -458,25 +474,28 @@ public void process(final ProcessorSupplier<K, V> processorSupplier, String... s
      * an operation that changes the key, i.e, selectKey, map(..), flatMap(..).
      * @param keySerde      Serdes for serializing the keys
      * @param valSerde      Serdes for serilaizing the values
+     * @param topicNamePrefix  prefix of topic name created for repartitioning
      * @return a new {@link KStreamImpl}
      */
     private KStreamImpl<K, V> repartitionForJoin(Serde<K> keySerde,
-                                                 Serde<V> valSerde) {
+                                                 Serde<V> valSerde,
+                                                 final String topicNamePrefix) {
 
-        String repartitionedSourceName = createReparitionedSource(this, keySerde, valSerde);
+        String repartitionedSourceName = createReparitionedSource(this, keySerde, valSerde, topicNamePrefix);
```

Comment:
Actually I was thinking we can pass the left-join table's state store's name, i.e. the name is passed from the join processor to the merger processor and then to the re-partition processor. Note that for the `KTableKTableJoinMerger.view()`, we are also using `parent1`, i.e. the left join table's view function in case of a three-way join, for example:

```
table1.join(table2.join(table3))
```

So we are still only materializing table1/2/3, but not the intermediate join result. We can follow the same idea as we pass the store name as well. Does this make sense?